applications/ALEapplication/python_scripts/trilinos_mesh_solver_base.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Check that applications were imported in the main script
+KratosMultiphysics.CheckRegisteredApplications("ALEApplication", "TrilinosApplication")
+
+# Import applications
+import KratosMultiphysics.TrilinosApplication as TrilinosApplication
+
+# Other imports
+import KratosMultiphysics.mpi as KratosMPI
+import mesh_solver_base
+
+
+def CreateSolver(mesh_model_part, custom_settings):
+    return TrilinosMeshSolverBase(mesh_model_part, custom_settings)
+
+
+class TrilinosMeshSolverBase(mesh_solver_base.MeshSolverBase):
+    def __init__(self, mesh_model_part, custom_settings):
+        super(TrilinosMeshSolverBase, self).__init__(mesh_model_part, custom_settings)
+        self.print_on_rank_zero("::[TrilinosMeshSolverBase]:: Construction finished")
+
+    #### Public user interface functions ####
+
+    def AddVariables(self):
+        super(TrilinosMeshSolverBase, self).AddVariables()
+        self.mesh_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.PARTITION_INDEX)
+        self.print_on_rank_zero("::[MeshSolverBase]:: Variables ADDED.")
+
+    def AddDofs(self):
+        super(TrilinosMeshSolverBase, self).AddDofs()
+        self.print_on_rank_zero("::[MeshSolverBase]:: DOFs ADDED.")
+
+    #### Specific internal functions ####
+
+    def get_communicator(self):
+        if not hasattr(self, '_communicator'):
+            self._communicator = TrilinosApplication.CreateCommunicator()
+        return self._communicator
+
+    def print_on_rank_zero(self, *args):
+        KratosMPI.mpi.world.barrier()
+        if KratosMPI.mpi.rank == 0:
+            print(" ".join(map(str,args)))
+
+    #### Private functions ####
+
+    def _create_linear_solver(self):
+        import trilinos_linear_elastic_ml_solver
+        nit_max = 10000
+        linear_tol = 1e-5
+        linear_solver = trilinos_linear_elastic_ml_solver.MultilevelLinearSolver(linear_tol, nit_max)
```

Comment:
it would be best if you use the interface which passes a Parameters object (the factory) in creating the linear solver